concordium-contracts-common-derive/src/lib.rs

```diff
@@ -155,9 +155,88 @@ pub fn deserial_with_state_derive(input: TokenStream) -> TokenStream {
     unwrap_or_report(derive::impl_deserial_with_state(&ast))
 }
 
-/// Derive the `SchemaType` trait for a type.
-/// If the feature `build-schema` is not enabled this is a no-op, i.e., it does
-/// not produce any code.
+/// Derive the [`SchemaType`] trait for a type matching the implementation when
+/// deriving [`Serial`].
```

Comment:
Is the fact that it matches `Serial` relevant? And is it always the case?
To me, the important part is just that I can use JSON.

Reply:
I find it relevant. It is a good point with the JSON, although when we have generated contract clients, this is not the only use case.